### ISSUE TITLE
Processing speed slow

### ISSUE THREAD

**Issue by kollols** [JaidedAI/EasyOCR#203]
I have been trying to run it on my laptop(without gpu). It takes almost 10-15minutes to ocr an image and print the output.Is this normal?

**Comment by rkcosmos** [comment]
Does your image have a lot of text? You can also try increasing batch_size.

**Comment by kollols** [comment]
@rkcosmos I have attached the image which I am trying to ocr.I am running it in an ec2-instance, using the below command reader.readtext('eu-0011.png', detail = 0,batch_size=3).Yet it took 3minutes. If i wish to do a POC,3mins is a long time actually.

![eu-0011](https://user-images.githubusercontent.com/55937851/89253964-a81b5700-d63b-11ea-8169-1ae9c37e2939.png)

**Comment by rkcosmos** [comment]
At current state, we cannot do much faster than this. We plan to boost process speed later in phase 2. Actually if your image is without noise, uniform background and horizontal like this + you don't use GPU, this is the case where [tesseract](https://github.com/tesseract-ocr/tesseract) might be a better solution than us.

**Comment by kollols** [comment]
@rkcosmos ,thanks for the update.Will wait for spped update in phase 2.I tried with tesseract,but thing is the character accuracy of EasyOcr is better than Tesseract,i feel.Hence wanted to try with it.

**Comment by ghandic** [comment]
You may want to sub out the recognition for Tesseract but use craft for the detection. Also you can reduce image size before sending to craft. This would result in this image taking ~ 8-10 seconds. 

Not sure where in the API you'd do this as I have built a similar solution to run on CPU in < 10s seconds per page. Cannot be open sourced at the moment.

**Comment by rkcosmos** [comment]
@ghandic have u tried [DB](https://github.com/MhLiao/DB)? Their paper claims it's better.

**Comment by ghandic** [comment]
It's faster, about 4x but it's pretrained model is signs and labels etc. currently doing PoC for retrainibg on document text as it was very poor for the pretrained model. CRAFT on the other hand was good out of the box as a pretrained model.

**Comment by hpanwar08** [comment]
@ghandic Did you manage to train DB on document text? 
CRAFT misses out single characters in latin (https://github.com/clovaai/CRAFT-pytorch/issues/79) due to the training set they used, do you see the same problem in DB.

**Comment by ghandic** [comment]
Similar problem, we were using a tiny backbone though - shufflenet, MobileNet and smaller resnets as we're inferring on CPU so it's a time/acc trade off

**Comment by hpanwar08** [comment]
DB also misses out single characters on scanned documents?

**Comment by ghandic** [comment]
![image](https://user-images.githubusercontent.com/23500353/97554174-8529f480-1a2a-11eb-9019-65fd49bb1d90.jpeg)

You can see it struggles on smaller regions but overall still pretty good for speed ~ 1.5 seconds on CPU

**Comment by hpanwar08** [comment]
Hey @ghandic thanks for this. 1.5 second is really good. 
Can the model you trained be opensourced? If not, can you provide some info on the datasets you used for training DB.

**Comment by ghandic** [comment]
Unfortunately we can't but we only had a dataset of forms, paper and things found around the house then applied pixel level and spatial level augmentations.

We cleaned up the DB code quite a lot and have the option to use a lot of different cutting edge text detectors CRAFT, DB etc... built our own framework, all models making use of ONNX

**Comment by hpanwar08** [comment]
Thanks for the info.
